### ISSUE TITLE
Documentation

### ISSUE THREAD

**Issue by ShakenCodes** [rust-lang/rust-clippy#13822]
### Summary

Documentation of as_pointer_underscore claims it is available in 1.81.0, but it is not present in 1.83.0
https://rust-lang.github.io/rust-clippy/master/index.html#/as_pointer_underscore

### Reproducer

I tried this code:

```bash
% cargo clippy --version
clippy 0.1.83 (90b35a62 2024-11-26)
% cargo clippy --explain as_pointer_underscore
unknown lint: as_pointer_underscore
%
```

I expected to see this happen:
The explanation of the lint to be given. (According to the documentation, I should be able to do this a 1.81.0.)

Instead, this happened:
`unknown lint: as_pointer_underscore` reported.

### Version

```text
rustc 1.83.0 (90b35a623 2024-11-26)
binary: rustc
commit-hash: 90b35a6239c3d8bdabc530a6a0816f7ff89a0aaf
commit-date: 2024-11-26
host: aarch64-apple-darwin
release: 1.83.0
LLVM version: 19.1.1
```

**Comment by roberthree** [comment]
This lint was introduced with #13251, two weeks ago, so it's not part of [1.83.0](https://github.com/rust-lang/rust-clippy/tree/rust-1.83.0), and certainly not [1.81.0](https://github.com/rust-lang/rust-clippy/tree/rust-1.81.0). The `1.81.0` version is probably due to the fact that the PR was opened much earlier. On the other hand, the version of newly added lints is corrected in new releases, as described in [4. Update `clippy::version` attributes](https://doc.rust-lang.org/nightly/clippy/development/infrastructure/changelog_update.html#4-update-clippyversion-attributes).

Since you are using the `1.83.0` version, you should look for lints in <https://rust-lang.github.io/rust-clippy/rust-1.83.0/index.html>, which does not contain `as_pointer_underscore`, instead of <https://rust-lang.github.io/rust-clippy/master/index.html>. In general, one can start from <https://rust-lang.github.io/rust-clippy/> and choose the correct version.

I have also encountered this problem, especially when clicking on links from search results for Clippy Lints. Perhaps there should be a noticeable difference between [Clippy Lints (stable)](https://rust-lang.github.io/rust-clippy/stable/index.html) and [Clippy Lints (master)](https://rust-lang.github.io/rust-clippy/master/index.html), apart from the different URL. Much like the difference between [std (stable)](https://doc.rust-lang.org/stable/std/) and [std (internal)](https://stdrs.dev/nightly/x86_64-unknown-linux-gnu/std/index.html).

**Comment by GuillaumeGomez** [comment]
You mean between [std (stable)](https://doc.rust-lang.org/stable/std/) and [std (nightly)](https://doc.rust-lang.org/nightly/std/)? 😉 

There is no difference except for the URL and the version displayed in the sidebar.

**Comment by roberthree** [comment]
That's true, stable and nightly is probably a better comparison. However, the reason I chose the "internal" documentation as an example was because the potentially wrong version numbers of new lints feel like an internal agreement that may not be obvious to users? I don't know if such internal agreements exist in nightly Rust, too. How does Clippy relate to the nightly Rust builds?
I understand that it might be complicated to introduce some difference between the master and stable webpages, since some master will be the next stable, but since new lints are unfinished anyway (as far as their version number is concerned), could "unfinished" lints be highlighted differently on the webpage?
